### PR TITLE
Fix summary report

### PR DESCRIPTION
**Pull request by KasiaKoz**
Resolves https://github.com/arup-group/genet/issues/180 by making the summary of access tags more general and reports them by (access) mode.
Also changed the keys to match with other genet reports and added summary report to run with validation in genet cli.

Changes to a previous test fixture surface the error seen in the issue. I thought there is no point to create a new fixture for this problem since it's not really an edge case, it represents what happens in practice really well, but I can do if you disagree.
Screenshot 2023-11-01 at 10 41 20

### CODE REVIEW

**Review by mfitz — APPROVED** [review]
Looks good. I've made a minor comment about key names.

File: tests/test_core_network.py [hunk 1]
```diff
@@ -5681,25 +5686,41 @@ def test_generating_summary_report(network_for_summary_stats):
...
+                "car_distance_catchment_tag",
             },
             "accessibility_tags": {
-                "Stops with tag bikeAccessible": 1,
-                "Unique values for bikeAccessible tag": {"true"},
-                "Stops with tag carAccessible": 1,
-                "Unique values for carAccessible tag": {"true"},
+                "bike": {
+                    "access_tag": "bikeAccessible",
+                    "number_of_stops_with_bikeAccessible_tag": 1,
```
[the remark this answers is not in the document]
Comment (KasiaKoz) [reply tests/test_core_network.py:5712]: ah yes agreed, great catch, I changed `number_of_stops_with_accessLinkId_mode_tag` to `number_of_stops_with_link_access_tag` last minute already and missed that one
